test/enumspecindex.test.ts

```diff
@@ -0,0 +1,57 @@
+import {EnumSpecIndex} from '../src/enumspecindex';
+import {SHORT_ENUM_SPEC} from '../src/enumspec';
+import {Property} from '../src/property';
+
+import {assert} from 'chai';
+
+describe('enumspecindex', () => {
+  describe('isEmpty', () => {
+    it('should return false if encoding property is set', () => {
+      let enumSpecIndex = new EnumSpecIndex()
+        .setEncodingProperty(0, Property.SCALE, SHORT_ENUM_SPEC);
```

Comment:
I don't know why TS doesn't complain yet, but we never set these to SHORT_ENUM_SPEC and it doesn't match the type.  Better fix them with full enum spec; otherwise, it might throw error when we upgrade to future version of TS.